R/matching.R

```diff
@@ -0,0 +1,140 @@
+#' Relaxed Value Matching
+#'
+#' These functions offer relaxed matching of one vector in another.
+#' In contrast to the similar [`match`] and [`%in%`] functions they
+#' just accept `numeric` arguments but have an additional `tolerance`
+#' argument that allows relaxed matching.
+#'
+#' @param x `numeric`, the values to be matched.
+#' @param table `numeric`, the values to be matched against. In contrast to
+#' [`match`] `table` has to be sorted in increasing order.
+#' @param tolerance `numeric`, accepted tolerance. Use `0` to get the
+#' same behaviour as [`match`]. Use `Inf` (default) to match to the closest
+#' value without restrictions. Could be of length one or the same length as
+#' `table`.
+#' @param duplicates `character(1)`, it is possible that many `x` match one
+#' element in `table`. For `duplicates="keep"` (default) all of them are
+#' returned. For `duplicates="closest"` just the `x` that is closest to the
+#' element in `table` gets an index (all the other will get `nomatch`). If the
+#' difference is exactly identical the first one in `x` will get an index. For
+#' `duplicates="remove"` all duplicated matches are returned as `nomatch`. If
+#' a single element in `x` matches to multiple elements in `table` the *closest*
+#' is returned for `duplicates="keep"` or `duplicates="duplicates"` (*keeping*
+#' multiple matches isn't possible in this case because the implentation relies
+#' on [`findInterval`]). If the difference is identical the index of the
+#' smaller element in `table` is returned.
+#' For `duplicates="remove"` all multiple matches are returned as `nomatch`.
+#' @param nomatch `numeric(1)`, if the difference
+#' between the value in `x` and `table` is larger than
+#' `tolerance` `nomatch` is returned.
+#'
+#' @return `closest` returns an `integer` vector of the same length as `x`
+#' giving the closest position in `table` of the first match or `nomatch` if
+#' there is no match.
+#'
+#' @rdname matching
+#' @author Sebastian Gibb
+#' @seealso [`match`]
+#' @aliases closest
+#' @export
+#' @examples
+#' ## Define two vectors to match
+#' x <- c(1, 3, 5)
+#' y <- 1:10
+#'
+#' ## Compare match and closest
+#' match(x, y)
+#' closest(x, y)
+#'
+#' ## If there is no exact match
+#' x <- x + 0.1
+#' match(x, y) # no match
+#' closest(x, y)
+#'
+#' ## Some new values
+#' x <- c(1.11, 45.02, 556.45)
+#' y <- c(3.01, 34.12, 45.021, 46.1, 556.449)
+#'
+#' ## Using a single tolerance value
+#' closest(x, y, tolerance = 0.01)
+#'
+#' ## Using a value-specific tolerance accepting differences of 20 ppm
+#' closest(x, y, tolerance = x * ppm(20))
```

Comment:
Shouldn't that be `tolerance = y * ppm(20)`? - length of tolerance should be either 1 or equal to `length(table)`

Reply:
You are right. That is left over from my initial attempt to change that according to your `matchApprox` implementation.